docs/key-issues/gpus.md

```diff
@@ -180,3 +204,84 @@ Potential solutions on the PyPI side include:
+version of the CUDA toolkit will be compatible with any driver version within
+the same major release. This behavior is useful because it is often easier for
+users to upgrade their CUDA runtime than it is to upgrade the driver,
+especially on shared machines. For instance, the CUDA toolkit may be installed
+using conda, while the driver library cannot be. An example of leveraging MVC
+would be compiling code against the CUDA 11.5 runtime library and then running
+it on a system with a CUDA 11.2 driver installed. However there are some
+caveats with MVC:
+
+- If the code uses any features that were introduced in a later driver version
+  than the installed version, it will still fail to run. However, it will be a
+  runtime failure in the form of a `cudaErrorCallRequiresNewerDriver` CUDA
+  error, rather than a linker error or some similarly opaque issue. One
+  solution to this problem is for libraries to use runtime checks of the CUDA
+  version (using e.g. `cudaDriverGetVersion`) to only use supported features on
+  the installed driver.
+- NVRTC did not start supporting MVC until CUDA 11.3. Therefore, code that uses
+  NVRTC for JIT compilation must have been compiled with a CUDA version >= 11.3.
+- MVC only applies to CUDA code, not
+  [PTX](https://docs.nvidia.com/cuda/parallel-thread-execution/). PTX is an
+  instruction set that the CUDA driver library can JIT-compile to machine
+  instructions. The standard [CUDA compilation
+  pipeline](https://docs.nvidia.com/cuda/cuda-compiler-driver-nvcc/index.html#the-cuda-compilation-trajectory)
+  includes the translation of CUDA code into PTX, but in addition developers
+  may write PTX code directly rather than CUDA for performance since it offers
+  avenues for optimization that are either unavailable or very difficult to
+  access with CUDA C(++). However, since PTX code does not support MVC, PTX
+  compiled with a particular CUDA runtime may not work if run on a system with
+  an older driver. This fact has two consequences. First, libraries that
+  package PTX code will not benefit from MVC. Second, libraries that leverage
+  any sort of JIT-compilation pipeline that generates PTX code will _also_ not
+  support MVC. The latter can lead to more surprising behaviors, such as if a
+  user has a newer CUDA runtime than driver and then uses
+  [numba.cuda](https://numba.pydata.org/) to compile a Python function since
+  numba compiles CUDA kernels to PTX as part of its pipeline. Prior to CUDA
+  12, CUDA itself provides no solutions to this problem, although in some cases
+  there are tools that may be used to help (for instance, numba supports MVC in
+  CUDA 11 [starting with numba
+  0.57](https://numba.readthedocs.io/en/stable/release-notes.html#version-0-57-0-1-may-2023).
+  CUDA 12 introduces the
+  [nvJitLink](https://docs.nvidia.com/cuda/nvjitlink/index.html) library as the
+  long-term solution to this problem. nvJitLink may be leveraged to compile PTX
```

Comment:
It does, but the Python binding is still WIP (https://github.com/gmarkall/nvjitlink) and Numba and any other applications will need modification to use it.

Reply:
I think we can probably leave that link out for now and simply state that Python bindings do not exist yet, then update the page when they do. Otherwise we'll be sending readers down a potentially unhelpful rabbit hole, especially if the bindings move into a different org or library over time.

Reply:
nvJitLink is the ultimate solution, but with its own caveats. To link together different pieces generated from different minor CTK versions, it requires to use nvJitLink coming from a CTK whose minor version is higher than those of all participating pieces (see the [CUDA minor version compatibility](https://developer.nvidia.com/blog/cuda-12-0-compiler-support-for-runtime-lto-using-nvjitlink-library/) section; need to search in the article, as there's no direct hyperlink, unfortunately). Ex: if I'm linking pieces generated from CTK 12.0, 12.1, and 12.2, I need to use nvJitLink from 12.2+.

Reply:
> I think we can probably leave that link out for now and simply state that Python bindings do not exist yet, then update the page when they do. Otherwise we'll be sending readers down a potentially unhelpful rabbit hole, especially if the bindings move into a different org or library over time.

Indeed, I didn't intend to imply we should add that link - just pointing to where the current state of things is :-)